internal/services/extendedlocation/extended_location_custom_location.go

```diff
@@ -0,0 +1,327 @@
+	arckubernetes "github.com/hashicorp/go-azure-sdk/resource-manager/hybridkubernetes/2021-10-01/connectedclusters"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type CustomLocationResource struct{}
+
+type CustomLocationResourceModel struct {
+	Name                string      `tfschema:"name"`
+	ResourceGroupName   string      `tfschema:"resource_group_name"`
+	Location            string      `tfschema:"location"`
+	Authentication      []AuthModel `tfschema:"authentication"`
+	ClusterExtensionIds []string    `tfschema:"cluster_extension_ids"`
+	DisplayName         string      `tfschema:"display_name"`
+	HostResourceId      string      `tfschema:"host_resource_id"`
+	HostType            string      `tfschema:"host_type"`
+	Namespace           string      `tfschema:"namespace"`
+}
+
+type AuthModel struct {
+	Type  string `tfschema:"type"`
+	Value string `tfschema:"value"`
+}
+
+func (r CustomLocationResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringMatch(
+				regexp.MustCompile(`^[A-Za-z\d.\-_]*[A-Za-z\d]$`),
+				"supported alphanumeric characters and periods, underscores, hyphens. Name should end with an alphanumeric character.",
+			),
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"location": commonschema.Location(),
+
+		"namespace": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringMatch(
+				regexp.MustCompile("^[a-zA-Z0-9][a-zA-Z0-9-.]{0,252}$"),
+				"namespace must be between 1 and 253 characters in length and may contain only letters, numbers, periods (.), hyphens (-), and must begin with a letter or number.",
+			),
+		},
+
+		"cluster_extension_ids": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+			},
+		},
+
+		"host_resource_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: arckubernetes.ValidateConnectedClusterID,
+		},
+
+		"host_type": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(customlocations.HostTypeKubernetes),
+			}, false),
+		},
+
+		"authentication": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"type": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+
+					"value": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
We'll need better validation here, can we chase the service team on your enquiry above?